core/include/scipp/core/string.h

```diff
@@ -98,6 +96,37 @@ SCIPP_CORE_EXPORT std::string
 format_variable(const std::string &key, const VariableConstView &variable,
                 const Dimensions &datasetDims = Dimensions());
 
-} // namespace scipp::core
+/// Abstract base class for formatters for variables with element types not in
+/// scipp-core module.
+class SCIPP_CORE_EXPORT AbstractFormatter {
+public:
+  virtual ~AbstractFormatter() = default;
+  virtual std::string format(const VariableConstView &var) const = 0;
+};
+
+/// Concrete class for formatting variables with element types not in
+/// scipp-core.
```

Comment:
Looks good. My only suggested improvement is that the concrete Formatter could be defined as a local class within a make function. Apart from some syntactic sugar would only make a minor difference here, but would give you template argument deduction if you needed it elsewhere.